Nether and End in settings.ini by default, and created at startup

SafwatHalaby opened the pull request:
Fixes part 1 of #2810

SafwatHalaby commented:
Apparently this is not enough.

SafwatHalaby commented:
Fixed and rebased.

worktycho commented on src/Root.cpp:
```diff
@@ -143,13 +143,20 @@ void cRoot::Start(std::unique_ptr<cSettingsRepositoryInterface> a_OverridesRepo)
+
 	if (!IniFile->ReadFile("settings.ini"))
 	{
+		NewIniFile = true;
```
I feel this might be better written as:
```
bool NewIniFile = !IniFile->ReadFile("settings.ini");
if (NewIniFile)

SafwatHalaby replied:
Fixed.

SafwatHalaby commented:
I think that before deprecating CreateAndInitializeWorld, we should make sure all those who have generated their world prior to this PR get some treatment. Otherwise many will end up without a Nether and an End without any obvious reason to them.

madmaxoft commented:
The code looks okay to me, except that I'd prefer naming bool values `IsXYZ`, `HasXYZ` etc. instead of simple `XYZ` - to make it clear that it's a flag, not the contents.

I assume you've tested this on a clean config?

There's still the matter of handling the linkage to non-existent worlds, but we can address that in a separate PR.

SafwatHalaby commented:
I renamed that variable and rebased.

Yes, I have tested that in various configs.
Before throwing errors on linkages to non-existent worlds, we must fix the servers who generated the world before this PR using this procedure:

If:
 - These folders exist: `world`, `world_nether`, `world_end` &&
 - `settings.ini` contains `world` but doesn't contain `world_nether` or `world_end`

Then:
 - Add `world_nether` and `world_end` to the ini.

SafwatHalaby commented:
Not happy with this at all. I think multiworld teleportation is too unstable and shouldn't be the default.

SafwatHalaby commented:
There shouldn't be a default Nether and End.

SafwatHalaby commented:
Perhaps the Nether and End shouldn't be traversable at all by default. Discussion here:
http://forum.mc-server.org/showthread.php?tid=2272

tigerw commented:
Hold on, my original intent was to have the format be \*current world name\*_end/_nether, rather than a preprogrammed world(_end/_nether) prefix.

SafwatHalaby commented:
But the code I added is triggered when NO INI FILE is present, as the default.
